app/core/redis.py

```
@@ -128,6 +146,46 @@ async def _health_check(self):
             self.is_healthy = False
             await self._handle_connection_failure()
     
+    async def _recreate_client_on_error(self):
+        """Recreate Redis client when parser errors occur."""
+        try:
+            # Close current client safely
+            if self.client:
+                try:
+                    await self.client.aclose()
+                except Exception:
+                    pass
```

Ruff (0.12.2)

`try`-`except`-`pass` detected, consider logging the exception

(S110)